### PR TITLE
Add Goodreads presence!

### LINKED ISSUE #2247: Goodreads | goodreads.com
**Prerequisites and essential questions** 
- [x] Is it a popular site?
- [x] Is the website older than 2 months? 
- [ ] Is the site locked to a specific country/region?
- [ ] Is the site a paid service? (e.g. Netflix, Hulu)
- [ ] Does the website feature NSFW content? (e.g. porn, etc...)
- [ ] Are you a donator/patron?
- [x] Do you acknowledge that coding presences is completely voluntary and may take time for your service to be added regardless of priority?

**What's your Discord username?**  jeijeisan#5105

**What's the name of the service?**  Goodreads | https://www.goodreads.com

**What should the Presence display?**  Playing/Browsing Goodreads, Browsing Home, Browsing [book title and author], Searching for..., Browsing profile

**If possible, please provide a logo for the service (512x512 minimum)**

### PR DESCRIPTION
**Pull request by ThiaudioTT**
Heyo.

I made a Goodreads presence 👍

<details>
  <summary>Screenshots</summary>

Screenshot_1

Screenshot_2

Screenshot_3

Screenshot_4

Screenshot_5

Screenshot_6

Screenshot_7

Screenshot_8

Screenshot_9

Screenshot_10

Screenshot_11

Screenshot_12

Screenshot_13

Screenshot_14

Screenshot_15

Screenshot_16

Screenshot_17

Screenshot_18

</details>
fixes #2247

### CODE REVIEW

**Comment by ThiaudioTT** [comment]
35bc1b96d81a13860d8e57436274ba6f3cbee5b0

i forgor about private profiles, but now are solved:
<details>
<summary>Print</summary>

Screenshot_20

</details>

File: websites/G/Goodreads/presence.ts [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+  };
+
+  if (document.location.pathname === "/home" ||
+      document.location.pathname === "/" ) 
+    presenceData.details = "Browsing homepage";
+  else if (document.location.pathname === "/book") 
+    presenceData.details = "Browsing books";
+  else if (document.location.pathname.includes("/book/show/")) {
+    // beta layout conditional
+    presenceData.details = "Browsing a book:";
```
Comment (SlowLife1661) [inline websites/G/Goodreads/presence.ts:21]:
```suggestion
    presenceData.details = "Viewing a book:";
```

File: websites/G/Goodreads/dist/metadata.json [hunk 2]
```diff
@@ -0,0 +1,20 @@
+{
+  "$schema": "https://schemas.premid.app/metadata/1.3",
+  "service": "Goodreads",
+  "author": {
+    "name": "ThiaudioTT3.2V6BITURBO",
+    "id": "867435320219533363"
+  },
+  "category": "other",
+  "logo": "https://i.imgur.com/DtHgzGt.png",
+  "thumbnail": "https://s.gr-assets.com/assets/facebook/goodreads_wide-e23f6858b6bf20dcaf8493237a214a0e.png",
```
Comment (SlowLife1661) [inline websites/G/Goodreads/dist/metadata.json:10]: Upload to imgur

File: websites/G/Goodreads/presence.ts [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+  } else if (document.location.pathname.includes("/series")) {
+    const bookseries: string = document.querySelector("h1").innerText;
+    if ( bookseries === "Series") 
+      presenceData.details = "Viewing all book series on Goodreads";
+    else {
+      presenceData.details = "Viewing a book series:";
+      presenceData.state = bookseries;
+    }
+  } else if (document.location.pathname.includes("/user/show/")) {
+    presenceData.details = "Browsing a profile:";
```
Comment (SlowLife1661) [inline websites/G/Goodreads/presence.ts:40]:
```suggestion
    presenceData.details = "Viewing a profile:";
```

File: websites/G/Goodreads/presence.ts [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+let book: string,
+  author: string;
+ 
+presence.on("UpdateData", async () => {
+
+  const presenceData: PresenceData = {
+    largeImageKey: "goodreads",
+    details: "Browsing Goodreads",
+    startTimestamp: Math.floor(Date.now() / 1000)
+  };
```
Comment (SlowLife1661) [inline websites/G/Goodreads/presence.ts:12]:
```suggestion
let book: string,
  author: string,
  timeElapsed = Math.floor(Date.now() / 1000);
 
presence.on("UpdateData", async () => {

  const presenceData: PresenceData = {
    largeImageKey: "goodreads",
    details: "Browsing Goodreads",
    startTimestamp: timeElapsed
  };
```

File: websites/G/Goodreads/presence.ts (hunk 1 above)
Comment (ThiaudioTT) [reply websites/G/Goodreads/presence.ts:21]: Thanks!

File: websites/G/Goodreads/presence.ts [hunk 5]
```diff
@@ -0,0 +1,85 @@
+const presence = new Presence({clientId: "867525909204566056"});
+
+let book: string,
+  author: string,
+  timeElapsed: number = Math.floor(Date.now() / 1000);
```
Comment (SlowLife1661) [inline websites/G/Goodreads/presence.ts:5]:
```suggestion
const presence = new Presence({clientId: "867525909204566056"}),
  timeElapsed: number = Math.floor(Date.now() / 1000);

let book: string,
  author: string;
```

**Review by Bas950 — CHANGES_REQUESTED** [review]
use .textContent instead of .innerText

**Comment by Isa27-0** [comment]
Presence Request that must be linked: #2247

**Comment by ThiaudioTT** [comment]
https://github.com/PreMiD/Presences/pull/4155#issuecomment-888397267
> Presence Request that must be linked: #2247


Thanks!
fix #2247

**Review by Bas950 — APPROVED** [review]

**Review by Slowlife01 — APPROVED** [review]

**Pull request merged**